presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestReorderWindows.java

```diff
@@ -215,23 +215,31 @@ public void testPrefixOfPartitionComesFirstRegardlessOfTheirOrderInSQL()
     }
 
```

Comment:
Instead of removing the test for lag function, could you please keep it but just change the expected result? Similarly as you did for `TestMergeWindows`.
It'd be in order to prevent regressions (i.e. when `SwapAdjacentWindowsBySpecification` got changed breaking the `pullAboveProjections` feature).
I'd keep both tests: `lag` function test which shows that windows with lag function are reordered correctly, and one with filter to show when we don't reorder.